ios/Classes/TiAirshipEventEmitter.m

```diff
@@ -0,0 +1,129 @@
+/* Copyright Airship and Contributors */
+
+#import "TiAirshipEventEmitter.h"
+#import "TiProxy.h"
+#import "TiAirshipEvent.h"
+#import "TiApp.h"
+
+@interface TiAirshipEventListeners : NSObject
+@property (nonatomic, weak) TiProxy *proxy;
+@property (nonatomic, strong) NSMutableDictionary *listeners;
+
+- (void)addListenersrForEvent:(NSString *)event
+                        count:(NSUInteger)count;
+
+- (void)removeListenersrForEvent:(NSString *)event
+                           count:(NSUInteger)count;
+
+- (BOOL)fireEvent:(id<TiAirshipEvent>)event;
+
+@end
+
+
+@implementation TiAirshipEventListeners
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        self.listeners = [NSMutableDictionary dictionary];
+    }
+    return self;
+}
+- (void)addListenersrForEvent:(NSString *)event
+                        count:(NSUInteger)count {
+    NSUInteger current = [self.listeners[event] unsignedIntegerValue];
+    current += count;
+    self.listeners[event] = @(current);
+}
+
+- (void)removeListenersrForEvent:(NSString *)event
+                        count:(NSUInteger)count {
+    NSUInteger current = [self.listeners[event] unsignedIntegerValue];
+    current -= count;
```

Comment:
its an unsigned integer

Reply:
I think that is the interpretation. It can still go below zero, it just wraps around: https://stackoverflow.com/questions/11771710/why-how-could-nsuinteger-be-returning-a-negative-number